### PR TITLE
gh-93925: Improve doc clarity of sqlite3 commit/rollback, and close

### LINKED ISSUE #93925: Improve docs/docstrings of sqlite3 commit()/rollback() and close()
_Originally posted by @mwichmann in https://github.com/python/cpython/pull/93890#discussion_r899641021_:

> So (real question, no criticism): does that mean there ought to be more detail in the commit/rollback descriptions, rather than the context manager bit?

The wording of the docs and docstrings for the commit, rollback, and close connection methods can be improved, in order to provide more clarity around transaction handling.

Fixed with:

- #93926 
- #94010
- #94011

### PR DESCRIPTION
**Pull request by erlend-aasland**

### CODE REVIEW

**Comment by erlend-aasland** [comment]
Followup of #93890

**Comment by erlend-aasland** [comment]
cc. @python/proofreaders

File: Doc/library/sqlite3.rst [hunk 1]
```diff
@@ -423,21 +423,20 @@ Connection Objects
 
    .. method:: commit()
 
-      This method commits the current transaction. If you don't call this method,
-      anything you did since the last call to ``commit()`` is not visible from
-      other database connections. If you wonder why you don't see the data you've
-      written to the database, please check you didn't forget to call this method.
```
Comment (erlend-aasland) [inline Doc/library/sqlite3.rst:429]: IMO, the original is way too verbose. I'm not sure we need the tongue-in-cheek parts on basic transaction handling. Can we assume that a person using the sqlite3 module has a minimum understanding of basic database concepts, such as transactions? I believe so. Maybe I'm wrong.

File: Doc/library/sqlite3.rst [hunk 2]
```diff
@@ -423,21 +423,20 @@ Connection Objects
...
 
    .. method:: close()
 
-      This closes the database connection. Note that this does not automatically
-      call :meth:`commit`. If you just close your database connection without
-      calling :meth:`commit` first, your changes will be lost!
+      Close the database connection.
+      Any pending transaction is not committed implicitly.
+      Make sure to :meth:`commit` before closing,
+      to avoid loosing pending changes.
```
Comment (bskinn) [inline Doc/library/sqlite3.rst]: "losing"

File: Doc/library/sqlite3.rst (hunk 1 above)
Comment (bskinn) [reply Doc/library/sqlite3.rst:429]: Seems reasonable to me 👍

File: Doc/library/sqlite3.rst [hunk 3]
```diff
@@ -423,21 +423,20 @@ Connection Objects
...
-      other database connections. If you wonder why you don't see the data you've
-      written to the database, please check you didn't forget to call this method.
+      Commit any pending transaction to the database.
+      If there is no open transaction, this method is a no-op.
 
    .. method:: rollback()
 
-      This method rolls back any changes to the database since the last call to
-      :meth:`commit`.
+      Roll back to the start of any pending transaction.
```
Comment (bskinn) [inline Doc/library/sqlite3.rst:431]: Could there be more than one open transaction here?

If so, I would prefer "...to the start of any pending transaction(s)." here.

File: Doc/library/sqlite3.rst (hunk 2 above)
Comment (erlend-aasland) [reply Doc/library/sqlite3.rst]:
```suggestion
      to avoid losing pending changes.
```

File: Doc/library/sqlite3.rst (hunk 3 above)
Comment (erlend-aasland) [reply Doc/library/sqlite3.rst:431]: One one transaction per database connection.

File: Doc/library/sqlite3.rst (hunk 1 above)
Comment (CAM-Gerlach) [reply Doc/library/sqlite3.rst:429]: Agreed—this is a reference document, not a tutorial, so it should be concise and to the point.

File: Doc/library/sqlite3.rst [hunk 4]
```diff
@@ -423,21 +423,20 @@ Connection Objects
...
+      If there is no open transaction, this method is a no-op.
 
    .. method:: close()
 
-      This closes the database connection. Note that this does not automatically
-      call :meth:`commit`. If you just close your database connection without
-      calling :meth:`commit` first, your changes will be lost!
+      Close the database connection.
+      Any pending transaction is not committed implicitly.
+      Make sure to :meth:`commit` before closing,
```
Comment (CAM-Gerlach) [inline Doc/library/sqlite3.rst]:
```suggestion
      Any pending transaction is not committed implicitly;
      make sure to :meth:`commit` before closing
```

Explicit connect motivation with action and improve flow; elide spurious comma.

**Review by CAM-Gerlach — COMMENTED** [review]
LGTM, aside from a couple minor suggested textual tweaks. Thanks!

File: Doc/library/sqlite3.rst (hunk 1 above)
Comment (erlend-aasland) [reply Doc/library/sqlite3.rst:429]: Thanks both. I'll mark this as resolved.

**Review by CAM-Gerlach — APPROVED** [review]
LGTM now; thanks @erlend-aasland :heavy_check_mark:

**Review by pradyunsg — APPROVED** [review]
Looks good to me! ^>^

**Comment by mwichmann** [comment]
I haz no approval privileges, but I'll still say this is clearly an improvement :)

**Comment by erlend-aasland** [comment]
Thanks for the reviews. Highly appreciated.

**Pull request merged**